homeassistant/components/simplisafe/config_flow.py

```diff
@@ -115,32 +75,139 @@ async def async_step_user(
             errors = {"base": "unknown"}
 
         if errors:
-            return self._async_show_form(errors=errors)
+            return self.async_show_form(
+                step_id=error_step_id,
+                data_schema=error_schema,
+                errors=errors,
+                description_placeholders={CONF_USERNAME: self._username},
+            )
+
+        assert self._simplisafe
+
+        if self._simplisafe.auth_state == AuthStates.PENDING_2FA_SMS:
+            return await self.async_step_sms_2fa()
+
+        try:
+            async with async_timeout.timeout(DEFAULT_EMAIL_2FA_TIMEOUT):
+                while True:
+                    try:
+                        await self._simplisafe.async_verify_2fa_email()
```

Comment:
If there's a task that may take a long time to complete we recommend using the progress step flow api to be able to inform the user about the task that needs to complete before the flow can continue:

https://developers.home-assistant.io/docs/data_entry_flow_index#show-progress-and-show-progress-done

Reply:
That is a good suggestion actually, forget about that one

Reply:
Thanks, @MartinHjelmare! I'll get that in a follow-up PR.